src/test/java/org/opensearch/action/admin/indices/mapping/get/IndexAnomalyDetectorActionHandlerTests.java

```diff
@@ -703,4 +708,59 @@ public void testTenMultiEntityDetectorsUpdateExistingMultiEntityAd() throws IOEx
         assertTrue(value instanceof IllegalStateException);
         assertTrue(value.getMessage().contains("NodeClient has not been initialized"));
     }
+
+    @SuppressWarnings("unchecked")
+    public void testCreateAnomalyDetectorWithDuplicateFeatureAggregationNames() throws IOException {
+        Feature featureOne = TestHelpers.randomFeature("featureName", "test-1");
+        Feature featureTwo = TestHelpers.randomFeature("featureNameTwo", "test-1");
+        AnomalyDetector anomalyDetector = TestHelpers.randomAnomalyDetector(ImmutableList.of(featureOne, featureTwo));
+        SearchResponse detectorResponse = mock(SearchResponse.class);
+        int totalHits = 9;
+        when(detectorResponse.getHits()).thenReturn(TestHelpers.createSearchHits(totalHits));
+
+        // extend NodeClient since its execute method is final and mockito does not allow to mock final methods
```

Comment:
Can we use `doAnswer` like https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/task/ADTaskManagerTests.java#L289-L309 ?

Reply:
Still looking into this. Will update this comment. At the moment I haven't been able to use doAnswer to test the method and extending the NodeClient has been the only way thats worked for me. This is also how other methods in this class have been tested.

Reply:
Changed this to use doAnswer instead of extending nodeClient